### PR TITLE
Add convenient ArcContainer.listAll() methods

### PR DESCRIPTION
**Pull request by mkouba**
- i.e. programmatic lookup for io.quarkus.arc.All

### CODE REVIEW

**Comment by mkouba** [comment]
Previously you'd need to use something like `Arc.container().instance(new TypeLiteral<List<Service>>() {}, All.Literal.INSTANCE).get()` or even `Arc.container().instance(new TypeLiteral<List<InstanceHandle<Service>>>() {}, All.Literal.INSTANCE).get()` if you want to destroy the `@Dependent` beans correctly.

File: independent-projects/arc/runtime/src/main/java/io/quarkus/arc/ArcContainer.java [hunk 1]
```diff
@@ -134,6 +134,32 @@ public interface ArcContainer {
      */
     <T> InjectableInstance<T> select(TypeLiteral<T> type, Annotation... qualifiers);
 
+    /**
+     * List all beans matching the required type and qualifiers.
```
Comment (Ladicek) [inline independent-projects/arc/runtime/src/main/java/io/quarkus/arc/ArcContainer.java:138]: Maybe document that the list is sorted by bean priority?

**Review by Ladicek — APPROVED** [review]
LGTM.

Out of curiosity, is this basically identical to `InjectableInstance.select(...).handles()`?

File: independent-projects/arc/runtime/src/main/java/io/quarkus/arc/ArcContainer.java (hunk 1 above)
Comment (mkouba) [reply independent-projects/arc/runtime/src/main/java/io/quarkus/arc/ArcContainer.java:138]: Well, it has the same semantics as `io.quarkus.arc.All` which is also referenced in the `@see` tag. But ofc I can add it here as well..

**Comment by mkouba** [comment]
> Out of curiosity, is this basically identical to `InjectableInstance.select(...).handles()`?

More or less, but `handles()` returns an `Iterable` and not a `List`. Also you'd have to use the `TypeLiteral` and the `@All` qualifier for regular `select()`.

**Comment by Ladicek** [comment]
Ah ah, I see, so `ArcContainer.listAll(type, qualifiers)` is roughly identical to `InjectableInstance.select(List<type>, All.class + qualifiers).handles()`, got it. Thanks!

File: independent-projects/arc/runtime/src/main/java/io/quarkus/arc/ArcContainer.java [hunk 2]
```diff
@@ -134,6 +134,36 @@ public interface ArcContainer {
...
+     * <p>
+     * Instances of dependent scoped beans should be explicitly destroyed with {@link InstanceHandle#destroy()}.
+     * <p>
+     * The list of is sorted by {@link InjectableBean#getPriority()}. Higher priority goes first.
```
Comment (manovotn) [inline independent-projects/arc/runtime/src/main/java/io/quarkus/arc/ArcContainer.java:142]:
```suggestion
     * The list is sorted by {@link InjectableBean#getPriority()}. Higher priority goes first.
```

File: independent-projects/arc/runtime/src/main/java/io/quarkus/arc/All.java [hunk 3]
```diff
@@ -59,7 +59,7 @@
  * }
  * </pre>
  *
- * By default, the list of beans is sorted by {@link InjectableBean#getPriority()}. Higher priority goes first.
+ * The list of is sorted by {@link InjectableBean#getPriority()}. Higher priority goes first.
```
Comment (manovotn) [inline independent-projects/arc/runtime/src/main/java/io/quarkus/arc/All.java:62]:
```suggestion
 * The list is sorted by {@link InjectableBean#getPriority()}. Higher priority goes first.
```

**Review by manovotn — APPROVED** [review]
Looks good

**Pull request merged**